src/librustc/traits/fulfill.rs

```diff
@@ -475,7 +475,8 @@ fn process_predicate<'a, 'gcx, 'tcx>(
                         trait_ref_type_vars(selcx, data.to_poly_trait_ref(tcx));
                     Ok(None)
                 }
-                Ok(v) => Ok(v),
+                // FIXME: call ProjectionCache::complete when we can.
```

Comment:
This would basically be when the obligation forest decides we are done with a projection predicate, right? In other words, the FIXME is sort of in the wrong place? I would expect us to iterate over `outcome.completed` (elsewhere in `fulfill.rs`...) and invoke `complete()` from there.